src/main/java/com/almostreliable/merequester/requester/RequesterBlockEntity.java

```diff
@@ -179,31 +179,26 @@ private boolean handleRequests() {
         var tickRateModulation = TickRateModulation.IDLE;
         for (var i = 0; i < requestStatus.length; i++) {
             var state = requestStatus[i];
-            var result = handleRequest(i);
-            if (!Objects.equals(state, result)) {
+
+            // if we change multiple states per tick then the client will never see us in "missing ingredients"
+            // we can either special case that state, or only transition once per tick, and once-per-tick is
+            // significantly less code :)
```

Comment:
While I agree that this change would result in less code, it's not very efficient to schedule new operations to the next tick especially when exporting is involved. This makes the requester noticeably slower.

It's incorrect that the client will never see the "missing ingredient" state with the current behavior. You can just remove the necessary ingredients for a recipe that is configured in the requester and it will show you the missing state in the GUI. This is specified here: https://github.com/AlmostReliable/merequester/blob/1.19-forge/src/main/java/com/almostreliable/merequester/requester/status/PlanState.java#L33

Reply:
In the same setup to reproduce the issues caused by the singleton MissingState I was experiencing an issue where constantly missing an ingredient would result in the client feedback only showing Idle. I believe the reason for this is because right after transitioning from PlanState to MissingState, handle is then invoked again: https://github.com/AlmostReliable/merequester/blob/95a98aa7c9ce8103563509fc76a3eca7f3efda2a/src/main/java/com/almostreliable/merequester/requester/RequesterBlockEntity.java#L201 and we were immediately returning to PlanState. As I was writing this comment, I investigated more and now realize that this is actually a 2nd artifact of MissingState being a singleton and addressing that root issue would eliminate this behavior as well.